core/src/main/java/com/microsoft/applicationinsights/internal/channel/samplingV2/FixedRateSamplingTelemetryProcessor.java

```diff
@@ -0,0 +1,175 @@
+                put(traceTelemetryName, Class.forName("com.microsoft.applicationinsights.telemetry.TraceTelemetry"));
+            }};
+        }
+        catch (ClassNotFoundException e) {
+            InternalLogger.INSTANCE.trace("Unable to locate telemetry classes");
+        }
+    }
+
+    public String getExcludedTypesString() {
+        return excludedTypesString;
+    }
+
+    public String getIncludedTypesString() {
+        return includedTypesString;
+    }
+
+    /**
+     * This method takes a string of user specified excluded
+     * types and parses it to into set containing respective references
+     * @param value
+     */
+    public void setExcludedTypes(String value) {
+        this.excludedTypesString = value;
+        Set<Class> newExcludedTypesHashSet = new HashSet<Class>();
+        setIncludedOrExcludedTypes(value, newExcludedTypesHashSet);
+        excludedTypesHashSet = newExcludedTypesHashSet;
+    }
+
+    /**
+     * This method takes a string of user specified included
+     * types and parses it to into set containing respective references
+     * @param value
+     */
+    public void setIncludedTypes(String value) {
+        this.includedTypesString = value;
+        Set<Class> newIncludedTypesHashSet = new HashSet<Class>();
+        setIncludedOrExcludedTypes(value, newIncludedTypesHashSet);
+        includedTypesHashSet = newIncludedTypesHashSet;
+    }
+
+    private void setIncludedOrExcludedTypes(String value, Set<Class> typeSet) {
+        if (!StringUtils.isNullOrEmpty(value)) {
+            String[] splitList = value.split(listSeparator);
+            for (String item : splitList) {
+                item.trim();
+                if (!StringUtils.isNullOrEmpty(item) && allowedTypes.containsKey(item)) {
+                    typeSet.add(allowedTypes.get(item));
+                }
+            }
+        }
+    }
+
+    public double getSamplingPercentage() {
+        return samplingPercentage;
+    }
+
+    /**
+     * Sets the user defined sampling percentage
+     * @param samplingPercentage
+     */
+    public void setSamplingPercentage(String samplingPercentage) {
+        this.samplingPercentage = Double.valueOf(samplingPercentage);
+    }
+
+    /**
+     * This method determines if the telemetry needs to be sampled or not.
+     * @param telemetry
+     * @return
+     */
+    @Override
+    public boolean process(Telemetry telemetry) {
+
+        double samplingPercentage  = this.samplingPercentage;
+
+        if (!(telemetry instanceof SupportSampling)) {
+            return true;
+        }
+
+        if (!isSamplingApplicable(telemetry.getClass())) {
+            InternalLogger.INSTANCE.trace("Skip sampling since %s type is not sampling applicable", telemetry.getClass());
+            return true;
+        }
+
+        SupportSampling samplingSupportingTelemetry = ((SupportSampling)telemetry);
+        if (samplingSupportingTelemetry.getSamplingPercentage() != null) {
```

Comment:
Is this correct? If  a sampling percentage is available (i.e. different than null), then shouldn't we consider the telemetry for sampling?